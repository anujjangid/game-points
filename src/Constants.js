export const Points = [
  {
    game: 'A',
    value: 50,
    bonus: 200,
    bonusNumber: 3
  },
  {
    game: 'B',
    value: 30,
    bonus: 90,
    bonusNumber: 2
  },
  {
    game: 'C',
    value: 20
  },
  {
    game: 'D',
    value: 15
  },
  {
    game: 'E',
    value: 10
  }
];

export const KahootTitle = 'Kahoot! POINTS';
export const PlayerTitle = 'PLAYERS ITEMS';
export const ItemHeading = 'Item';
export const ItemQty = 'Qty';
export const ItemScore = 'Score';
export const NewGame = 'New Game';
export const BonusText = 'Bonus';
export const TotalText = 'Total';
export const StartNewGame = 'reset';
