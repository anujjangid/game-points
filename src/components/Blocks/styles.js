import styled from 'styled-components';

export const Wrapper = styled.section`
  display: flex;
  justify-content: space-around;
  padding: 20px 10px;
  flex-direction: column;

  align-items: center;
  @media only screen and (min-width: 768px) {
    flex-direction: row;
  }
`;

export const Block = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 70px;
  height: 70px;
  padding: 20px;
  border: 5px solid #ab3c3c;
  background: #f8f9e1;
  box-sizing: border-box;
  border-radius: 10px;
  color: #ab3c3c;
  font-size: 20px;
  cursor: pointer;
  transition: all 0.5s ease;
  margin-top: 20px;

  @media only screen and (min-width: 768px) {
    width: 100px;
    height: 100px;
    margin-top: 0px;
    font-size: 24px;
  }
  &:hover {
    font-size: 28px;
    border-color: black;
    color: #000;
    @media only screen and (min-width: 768px) {
      font-size: 48px;
    }
  }
`;
