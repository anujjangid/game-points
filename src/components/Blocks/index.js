import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { CounterContext } from '../../reducer';
import { Wrapper, Block } from './styles';

const Blocks = props => {
  const { dispatch } = useContext(CounterContext);
  const { data } = props;

  return (
    <Wrapper>
      {data.map((item, index) => {
        return (
          <Block key={index} onClick={() => dispatch({ type: item.game })}>
            {item.game}
          </Block>
        );
      })}
    </Wrapper>
  );
};

export default Blocks;

Blocks.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      bonus: PropTypes.number,
      bonusNumber: PropTypes.number,
      game: PropTypes.string.isRequired,
      value: PropTypes.number.isRequired
    })
  )
};
