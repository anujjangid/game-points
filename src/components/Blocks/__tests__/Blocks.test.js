import React from 'react';
import { create } from 'react-test-renderer';
import { Points } from '../../../Constants';
import Blocks from '../index';

describe('Test suite for Blocks Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions
  it('render correctly', () => {
    const component = create(<Blocks data={Points} />);
    expect(component.toJSON()).toMatchSnapshot();
    const rootInstance = component.root;
    const Block = rootInstance.findByType('section');
    expect(Block.props.children).toHaveLength(5);
  });
});
