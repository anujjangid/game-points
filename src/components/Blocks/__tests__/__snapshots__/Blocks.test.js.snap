// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Blocks Component render correctly 1`] = `
<section
  className="sc-bdVaJa kKLUAd"
>
  <div
    className="sc-bwzfXH kSOKvt"
    onClick={[Function]}
  >
    A
  </div>
  <div
    className="sc-bwzfXH kSOKvt"
    onClick={[Function]}
  >
    B
  </div>
  <div
    className="sc-bwzfXH kSOKvt"
    onClick={[Function]}
  >
    C
  </div>
  <div
    className="sc-bwzfXH kSOKvt"
    onClick={[Function]}
  >
    D
  </div>
  <div
    className="sc-bwzfXH kSOKvt"
    onClick={[Function]}
  >
    E
  </div>
</section>
`;
